.gitattributes

```diff
@@ -0,0 +1 @@
+*.txt linguist-language=AdBlock linguist-detectable
```

Comment:
@scripthunter7 I noticed that `[uBlock Origin]` can be added to the beginning of each list to get syntax highlighting there.

So can/should the `linguist-language` here be "uBlock Origin" as well?

Reply:
Yes, these use Finnish list.

Reply:
@MasterKia If the `[uBlock Origin]` agent is present at the beginning of every file, it is not necessary. 

But if you want statistics like this, then you need `linguist-detectable`
![image](https://user-images.githubusercontent.com/57285466/188590989-d481a29d-3841-4de6-a93d-9d7b51e9d347.png)

```gitattributes
*.txt linguist-detectable
```

Reply:
I meant something like `*.txt linguist-language="uBlock Origin"`

Reply:
> I meant something like `*.txt linguist-language="uBlock Origin"`

This is simply not possible as Linguist doesn't know such a language or alias.
See https://github.com/github/linguist/blob/bf853f1c663903e3ee35935189760191f1c45e1c/lib/linguist/languages.yml#L230-L242

Reply:
Also uBlock Origin is not an "**_ad_** blocker", so personally I don't like that statistics.

Reply:
> Also uBlock Origin [is not an "**_ad_** blocker"](https://github.com/gorhill/uBlock/#readme)

Neither is AdGuard, but their collective name is adblock

Reply:
AdGuard literally calls itself "Ad Blocker":
https://github.com/AdguardTeam/AdguardBrowserExtension#ad-blocker-with-advanced-privacy-protection-features

Reply:
> but I'm actually glad that we finally managed to add grammar to Linguist.

Thank you for the going through the heavy-lifting to add this feature.

Reply:
> but I'm actually glad that we finally managed to add grammar to Linguist.

Thank you for going through the heavy-lifting to add this feature.

Reply:
We can disable detect shell language in stats by same file.

Later you can same for Persian repo.

Reply:
Yes, but AdGuard can also block a lot of things, not just ads. Actually I don't want to do justice in this, I just wanted to help with the highlighter. 🙂 Perhaps you can try to [submit several aliases to Linguist](https://github.com/github/linguist/blob/bf853f1c663903e3ee35935189760191f1c45e1c/lib/linguist/languages.yml#L236-L240), eg. `adblock plus`, `ublock` / `ublock origin`, `adguard`, but the name of the language remains "Adblock Filter List" in that case as well.

Reply:
So this must wait for alias and disable generate stats as long don't count as other section "Wide spectrum filter list".

Reply:
> So this must wait

Or just add `[uBlock Origin]` to each file.

Reply:
As far as I know, the statistics always use the original names of the languages, i.e. in this case, even if you enter an alias in the .gitattributes file, the statistics will still show "Adblock Filter List"